### DISCUSSION TITLE
Enhancements request to MoveToResource and CodeReferences tools

### DISCUSSION THREAD

**Discussion by eng-marani** [dotnet/ResXResourceManager#554] [Ideas]
The regular expressio/string replacement for the tool, allow the use of some undocumented placeholders

$Key ---> resource key
$File ---> source file
$Namespace ---> resource namespace

I think there are some important keys that are missing (need to be implemented)
$Value ---> value of the resource (Neutral format)
$Class ---> name of the class where the MoveToResource is invoked

The first one ($Value) should be very useful to find any reference of that string into code.
Also for the MoveToResource tool, the $Value would be a nice addition when you have a custom function like this:
..... = "my resource string" ----> ..... = Translate.I18N("[token]","[default string value]");
There is actually no way to fetch the original string value (as default) and place it as parameter on the I18N() function.

Last but not the least, the MoveToResource dialog, if possible, should not be modal, but allow the user to Cut&Paste from the current source code, when the key is not arbitrary but must contain some source code data.

**Comment by tom-englert** [comment]
Happy to accept your PR, if you want to extend this.
